### ISSUE TITLE
Getting "CONNECTION_RESET" when trying to run containers

### ISSUE THREAD

**Issue by atrauzzi** [aspnet/aspnet-docker#106]
Just tried a new yeoman ASP.NET project and I'm using the dockerfiles `microsoft/aspnet:1.0.0-beta8` or `microsoft/aspnet:1.0.0-beta8-coreclr`.

In both cases, when I run the containers and I navigate to `http://localhost:5000`, Chrome tells me `ERR_CONNECTION_RESET`.

`dnx` seems to start up fine, and when I attach to the containers, it's still running on port 5000.  I've correctly exposed the ports as the message isn't saying that the connection is being refused.

Something seems amiss here as I have other containers that bind to ports just fine.  I haven't modified the project at all.  Everything is quite vanilla, just trying to see if it boots out of the box right now.  I'm on Ubuntu 15.10.

If I install links locally on the container and do `docker -it /bin/bash ...`, run it and navigate to `localhost:5000`, it seems to work.  Is there possibly an issue with kestrel running on docker?

**Comment by atrauzzi** [comment]
After a few more minutes of desperate badgering, it looks like my docker compose file needed `net: host`.  Seems to have sorted things out!

And that's my derp for the day.

**Comment by atrauzzi** [comment]
Now that I think about it, this solution is just running it directly on my network interface, and port 5000, not 8080 as I'm trying to remap it.  So I wonder if there are still issues?

**Comment by ahmetalpbalkan** [comment]
@atrauzzi You should not need --net=host. Just to make sure, this is not `yo docker` right?

Can you please send outputs of:

* `docker inspect [container]`
* `docker top [container] aux`
* `netstat -ntlp` 

in your Ubuntu host machine where the docker engine is running?

**Comment by atrauzzi** [comment]
Hey @ahmetalpbalkan !

The project was created using `yo aspnet`.  It came with a Dockerfile, so I'm working with it.  The only thing I've changed as I'm muddling through things right now is the port from 5000 to 8080 and I'm passing this parameter in my `project.json` as part of my web command: 

```
"commands": {
    "web": "Microsoft.AspNet.Server.Kestrel --server.urls http://*:8080/",
```

```
$ docker inspect aspnetgae_web
[
{
    "Id": "471c3da280eadfe0246b3c51c0b8561b2183b4f4b1e00084743d8e1eb2c3f430",
    "RepoTags": [
        "aspnetgae_web:latest"
    ],
    "RepoDigests": [],
    "Parent": "4fe93b34d53de0229a167cefcfe366366e2e606e4adb79c155f334b6329c8eab",
    "Comment": "",
    "Created": "2015-11-12T03:01:15.133095026Z",
    "Container": "ac0370c24a26b24a6e16bc9319d0a77317221c17658cd365e43a2d9008d22e5b",
    "ContainerConfig": {
        "Hostname": "f259a376263c",
        "Domainname": "",
        "User": "",
        "AttachStdin": false,
        "AttachStdout": false,
        "AttachStderr": false,
        "ExposedPorts": {
            "8080/tcp": {}
        },
        "Tty": false,
        "OpenStdin": false,
        "StdinOnce": false,
        "Env": [
            "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/dnx/runtimes/default/bin",
            "DNX_VERSION=1.0.0-beta8",
            "DNX_USER_HOME=/opt/dnx",
            "MONO_THREADS_PER_CPU=50"
        ],
        "Cmd": [
            "/bin/sh",
            "-c",
            "#(nop) ENTRYPOINT \u0026{[\"dnx\" \"-p\" \"project.json\" \"web\"]}"
        ],
        "Image": "4fe93b34d53de0229a167cefcfe366366e2e606e4adb79c155f334b6329c8eab",
        "Volumes": null,
        "WorkingDir": "/app",
        "Entrypoint": [
            "dnx",
            "-p",
            "project.json",
            "web"
        ],
        "OnBuild": [],
        "Labels": {}
    },
    "DockerVersion": "1.9.0",
    "Author": "",
    "Config": {
        "Hostname": "f259a376263c",
        "Domainname": "",
        "User": "",
        "AttachStdin": false,
        "AttachStdout": false,
        "AttachStderr": false,
        "ExposedPorts": {
            "8080/tcp": {}
        },
        "Tty": false,
        "OpenStdin": false,
        "StdinOnce": false,
        "Env": [
            "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/dnx/runtimes/default/bin",
            "DNX_VERSION=1.0.0-beta8",
            "DNX_USER_HOME=/opt/dnx",
            "MONO_THREADS_PER_CPU=50"
        ],
        "Cmd": null,
        "Image": "4fe93b34d53de0229a167cefcfe366366e2e606e4adb79c155f334b6329c8eab",
        "Volumes": null,
        "WorkingDir": "/app",
        "Entrypoint": [
            "dnx",
            "-p",
            "project.json",
            "web"
        ],
        "OnBuild": [],
        "Labels": {}
    },
    "Architecture": "amd64",
    "Os": "linux",
    "Size": 0,
    "VirtualSize": 1234516735,
    "GraphDriver": {
        "Name": "devicemapper",
        "Data": {
            "DeviceId": "273",
            "DeviceName": "docker-8:3-2493073-471c3da280eadfe0246b3c51c0b8561b2183b4f4b1e00084743d8e1eb2c3f430",
            "DeviceSize": "107374182400"
        }
    }
}
]
```

```
$ docker top aspnetgae_web_1 aux
USER                PID                 %CPU                %MEM                VSZ                 RSS                 TTY                 STAT                START               TIME                COMMAND
root                32576               2.2                 1.6                 239788              133276              ?                   Ssl                 21:01               0:02                mono /opt/dnx/runtimes/dnx-mono.1.0.0-beta8/bin/Microsoft.Dnx.Host.Mono.dll -p project.json web
```

```
$ netstat -ntlp
(Not all processes could be identified, non-owned process info
 will not be shown, you would have to be root to see it all.)
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 127.0.1.1:53            0.0.0.0:*               LISTEN      -               
tcp        0      0 127.0.0.1:631           0.0.0.0:*               LISTEN      -               
tcp6       0      0 ::1:631                 :::*                    LISTEN      -               
tcp6       0      0 :::32769                :::*                    LISTEN      -               
```

**Comment by ahmetalpbalkan** [comment]
It looks like you really intend to listen on 8080. However there are some things that I noticed in these outputs:

* `docker inspect` output does not container `NetworkSettings` key where port mappings are shown. I can hardly imagine any case this key is omitted from the output. What is your full `docker run` command to start this container?
* `netstat -ntlp` output shows that nothing is listening on 8080 or 5000. So this makes me think you have not mapped the container port to the host port using `docker run -p` (curl-ing inside container works, but not outside).

Therefore, providing your `docker run` command or the way you start this container can be very helpful.

**Comment by atrauzzi** [comment]
I'm running things via docker compose, here's the output of `docker-compose --verbose up`:

```
$ docker-compose --verbose up
Please be aware that .yml is the expected extension in most cases, and using .yaml can cause compatibility issues in future.

Compose version 1.4.2
Docker base_url: http+docker://localunixsocket
Docker version: KernelVersion=4.3.0-040300rc2-generic, Os=linux, BuildTime=Tue Nov  3 17:54:53 UTC 2015, ApiVersion=1.21, Version=1.9.0, GitCommit=76d6bc9, Arch=amd64, GoVersion=go1.4.2
docker containers <- (all=True, filters={u'label': [u'com.docker.compose.project=aspnetgae', u'com.docker.compose.service=web', u'com.docker.compose.oneoff=False']})
docker containers -> (list with 1 items)
docker inspect_container <- (u'9f27c0aa6db31b0cfcf25f2e60fe0a39d84357f7ff0b494d9f40cbda72e50638')
docker inspect_container -> {u'AppArmorProfile': u'',
 u'Args': [u'-p', u'project.json', u'web'],
 u'Config': {u'AttachStderr': False,
             u'AttachStdin': False,
             u'AttachStdout': False,
             u'Cmd': None,
             u'CpuShares': 0,
             u'Cpuset': u'',
             u'Domainname': u'',
             u'Entrypoint': [u'dnx', u'-p', u'project.json', u'web'],
...
docker containers <- (all=True, filters={u'label': [u'com.docker.compose.project=aspnetgae', u'com.docker.compose.service=web', u'com.docker.compose.oneoff=False']})
docker containers -> (list with 1 items)
docker inspect_image <- (u'aspnetgae_web')
docker inspect_image -> {u'Architecture': u'amd64',
 u'Author': u'',
 u'Comment': u'',
 u'Config': {u'AttachStderr': False,
             u'AttachStdin': False,
             u'AttachStdout': False,
             u'Cmd': None,
             u'Domainname': u'',
             u'Entrypoint': [u'dnx', u'-p', u'project.json', u'web'],
             u'Env': [u'PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/opt/dnx/runtimes/default/bin',
...
docker inspect_container <- (u'9f27c0aa6db31b0cfcf25f2e60fe0a39d84357f7ff0b494d9f40cbda72e50638')
docker inspect_container -> {u'AppArmorProfile': u'',
 u'Args': [u'-p', u'project.json', u'web'],
 u'Config': {u'AttachStderr': False,
             u'AttachStdin': False,
             u'AttachStdout': False,
             u'Cmd': None,
             u'CpuShares': 0,
             u'Cpuset': u'',
             u'Domainname': u'',
             u'Entrypoint': [u'dnx', u'-p', u'project.json', u'web'],
...
aspnetgae_web_1 is up-to-date
Attaching to aspnetgae_web_1
docker attach <- (u'9f27c0aa6db31b0cfcf25f2e60fe0a39d84357f7ff0b494d9f40cbda72e50638', stream=1, logs=1, stderr=1, stdout=1)
docker attach -> <generator object _multiplexed_response_stream_helper at 0x7f9477c6fb40>
web_1 | Could not determine OS version information. Defaulting to the empty string.
web_1 | info    : [Microsoft.Framework.DependencyInjection.DataProtectionServices] User profile is available. Using '/root/.local/share/ASP.NET/DataProtection-Keys' as key repository; keys will not be encrypted at rest.
web_1 | Hosting environment: Production
web_1 | Now listening on: http://*:8080/
web_1 | Application started. Press Ctrl+C to shut down.
```

My `docker-compose.yaml` is as follows:

```
web:
  build: .
  ports:
    - "8080"
```

(pretty simple ;)  )

**Comment by ahmetalpbalkan** [comment]
Your docker-compose.yaml is problematic, can you change ports to `"8080:8080"` to actually map port 8080 of the container to 8080 of the host machine?

Did you write this compose file or is it an artifact of `yo aspnet`?

**Comment by atrauzzi** [comment]
Ha, so the `8080:8080` did it!  For some reason I thought just putting the port on it's own brought it forward.  Clearly I was mistaken.

That compose file is my own as I'm experimenting with some things right now.

What is worth noting however is that I did have `"8080:8080"` in there earlier and it wasn't working (I've been changing lots of things this evening).  The line I added to make kestrel listen on `*` might still be needed here, which is part of the `yo aspnet` scaffolding.
